src/main/webapp/2Dscript.js

```diff
@@ -1,56 +1,122 @@
-// Gobal variables.
+// Global variables.
 let map;
+const mapCache = {};
+const dataCache = {};
 let pose;
-let runId;
-let dataType;
+let data;
+let id;
+let type;
+let currentLat;
+let currentLong;
+let initialPoseData;
```

Comment:
It's getting pretty difficult to follow the data flow with all the global variables. Can you make these into local variables passed around explicitly as much as possible?

Reply:
I'll work on moving some of these into functions. I know I was having a bit of a challenge figuring out the proper scope for variables passed to event handlers, any tips for that?

Reply:
I'm not sure what you mean. Are you talking about something like:
```
onclick(event) {
  map.setSomething(event.something) // map is currently global
}
```
and wondering what the scope of `map` should be? Or something else?

Reply:
Yeah so it gets confusing to me if something like this occurs
function doSomething(){
let a;
let b;
b.addEventListener('click', function(){
// reference the a which is the scope of doSomething
console.log(a);
});
vs if you factor out this lambda into a function, passing that value to the function without using global scope. I tried using bind, but I think I was misunderstanding how to use 'this' in this case?

Reply:
How about:
```
function doSomething() {
  let a;
  let b;
  b.addEventListener('click', function() { handleClick(a) });
}
function handleClick(a) {
  console.log(a);
}
```
There is still a lambda, but it's very short even if `handleClick` is long.

Reply:
for (int i =0 ; i <5; i++){
a.push(i)

Reply:
b[i].addEventListener